### DISCUSSION TITLE
multiple target(ee_link)?

### DISCUSSION THREAD

**Discussion by Spphire** [NVlabs/curobo#40] [Research Q&A] [answered]
Is it possible to set multiple target(ee_link) in mpc to avoid local minima to a certain extent?
e.g. one at end link and one at middle link

**Comment by balakumar-s** [comment]
That can be done, but only full pose can be set (not just position). This requires changes to a few files. Let me know if you want to try this.

Another option is to set a goal joint configuration in addition to the goal pose. This can be done by adding this line below:

https://github.com/NVlabs/curobo/blob/e0e804e9062da33c13aa262a1da253d364b8768a/examples/isaac_sim/mpc_example.py#L309

```
goal_cfg = torch.as_tensor([0,0,0,0,0,0,0],device="cuda").unsqueeze(0)
goal_buffer.goal_state.copy_(JointState.from_position(goal_cfg, joint_names=joint_names))
```

Change the values in `goal_cfg` to your target joint configuration.

**Reply by Spphire** [reply]
Does that mean while mpc following the end-link target, I can alse fix the angles of some of the joints in "joint_names"?

**Reply by balakumar-s** [reply]
You have to set joint angles for all joints. If you want to set only specific joints, you have to change the cost functions here: https://github.com/NVlabs/curobo/blob/e0e804e9062da33c13aa262a1da253d364b8768a/src/curobo/rollout/arm_reacher.py#L261

**Comment by Spphire** [comment]
What I mean is while mpc is following the end-link target(position and rotation), for example, the elbow can be following another target(position but rotation is not necessary).

**Reply by balakumar-s** [reply, marked as answer]
Yes, that's doable by changing the code here: https://github.com/NVlabs/curobo/blob/e0e804e9062da33c13aa262a1da253d364b8768a/src/curobo/rollout/arm_reacher.py#L244

But this is not doable from API.
